app/src/main/java/com/example/frolic/Event.java

```diff
@@ -1,43 +1,69 @@
 package com.example.frolic;
+import android.graphics.Bitmap;
 import android.util.Log;
 
-import com.google.firebase.firestore.auth.User;
-
 import java.util.Date;
 
 import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
 
 public class Event {
-    private Organizer organizer;
+    private String eventId;
+    private String organizerId;
     private String eventName;
     private String eventDesc;
-    private ArrayList<Entrant> confirmed;
+    private ArrayList<Entrant> confirmed = new ArrayList<>();
     private Facility facility;
-    private int maxEntrants;
+    private int waitlistLimit;
     private int maxConfirmed;
     private Date eventDate;
     private Date enrollDate;
     private final LotterySystem lottery;
+    private double price;
+    private boolean geolocationRequired;
+    private boolean receiveNotification;
+    private String qrHash;
 
     /**
      * This constructor acts as the main method of creating an event.
      * Note that Facility is not provided as it should be a prerequisite to create one before starting an event.
      * I am assuming that the UI will prompt the user to create a facility when trying to create an event if they do not already have one.
      */
-    public Event(Organizer organizer, String eventName, String eventDesc, int maxConfirmed, int maxEntrants, Date eventDate, Date enrollDate) {
+    public Event(String eventId, Organizer organizer, String eventName, String eventDesc, int maxConfirmed, int waitlistLimit, Date eventDate, Date enrollDate, double price, boolean geolocationRequired, boolean receiveNotification, String qrHash) {
         try { assert organizer.getFacility() != null; }
         catch (AssertionError e) { Log.e("Event.java", "Tried to create an event without a facility", e); }
 
-        this.organizer = organizer;
+        this.eventId = eventId;
+        this.organizerId = organizer.getOrganizerId();
         this.eventName = eventName;
         this.eventDesc = eventDesc;
         this.maxConfirmed = maxConfirmed;
-        this.maxEntrants = maxEntrants;
+        this.waitlistLimit = waitlistLimit;
         this.facility = organizer.getFacility();
-
         this.eventDate = eventDate;
         this.enrollDate = enrollDate;
-        lottery = new LotterySystem(this, maxEntrants, maxConfirmed);
+        this.price = price;
+        this.geolocationRequired = geolocationRequired;
+        this.receiveNotification = receiveNotification;
+        this.qrHash = qrHash;
+        lottery = new LotterySystem(this);
+    }
+
+    public Map<String, Object> toMap() {
```

Comment:
Do you know if you could write a javadoc for this? would like to see the use cases for this map and understand what it's for

Reply:
Sure! Essentially what this is doing is formatting the Event in a way that can put into Firebase. Firebase uses maps. But I'll add a Javadoc for sure, I actually meant to but must have missed that one.